sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -115,7 +120,40 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
         return reqBodyLoggingMono;
     }
 
-    private Function<HttpResponse, Mono<HttpResponse>> logResponseDelegate(final ClientLogger logger, final URL url,
+    private void formatAllowableHeaders(Set<String> allowedHeaderNames, HttpRequest request, ClientLogger logger) {
+        for (HttpHeader header : request.getHeaders()) {
+            if (allowedHeaderNames.contains(header)) {
+                logger.info(header.toString());
+            } else {
+                logger.info(REDACTED_PLACEHOLDER);
+            }
+        }
+    }
+
+    private void formatAllowableQueryParams(Set<String> allowedQueryParamNames, HttpRequest request,
+                                            ClientLogger logger) {
+        StringBuilder sb = new StringBuilder();
+        String queryString = request.getUrl().getQuery();
```

Comment:
I guess we need a null check for `queryString`, if i remember correctly `getQuery()` can return null if there is no query segment in the uri. just double check.